extension/httpfs/httpfs_extension.cpp

```diff
@@ -59,6 +60,10 @@ static void LoadInternal(DatabaseInstance &instance) {
 	provider->SetAll();
 
 	CreateS3SecretFunctions::Register(instance);
+
+	// set parquet_use_openssl to true when httpfs is loaded
+	config.options.parquet_use_openssl = true;
```

Comment:
Instead of setting this option here - perhaps Parquet can by default use `config.encryption_state` (i.e. have this setting default to true, where it tries to use `config.encryption_state` if available, and only when explicitly disabled do we not use that)?

Reply:
Very curious PR, expecially with the inter-extensions dependencies.

Would it not make sense to have the setting have 3 states, like:
* `SET parquet_encryption_mode = 'openssl';` -> needs HTTPFS extension loaded, errors out if no httpfs is there (possibly tries autoinstall + loading, maybe a follow up)
* `SET parquet_encryption_mode = 'mbedtls';` -> uses mbedtls backend
* `SET parquet_encryption_mode = 'auto';` -> uses openssl if available (but never changes set of extension loaded), otherwise mbedtls. This could be the default.

Reply:
I think this should really only be a debug setting, so I would not think too much about how to set this up. It should actually also be prefixed with `debug` imo

Reply:
I changed it now to this to still enable mbedtls:

For copy:
```
# write files with OpenSSL disabled (if httpfs is loaded)
COPY (SELECT 42 i) to '__TEST_DIR__/encrypted${key_len}_openssl.parquet' 
(ENCRYPTION_CONFIG {footer_key: 'key${key_len}'}, 
DEBUG_USE_OPENSSL false)
```

For read:
```
# read encrypted files by using mbedtls (if httpfs is loaded)
SELECT * FROM read_parquet('__TEST_DIR__/encrypted${key_len}_openssl.parquet', 
encryption_config={footer_key: 'key${key_len}'}, 
debug_use_openssl=false)
```